packages/dds/map/src/test/mocha/fuzzUtils.ts

```diff
@@ -0,0 +1,511 @@
+}
+
+/**
+ * Represents a map delete key operation.
+ */
+interface MapDeleteKey {
+	type: "deleteKey";
+	key: string;
+}
+
+type MapOperation = MapSetKey | MapDeleteKey | MapClear;
+
+// This type gets used a lot as the state object of the suite; shorthand it here.
+type MapState = DDSFuzzTestState<MapFactory>;
+
+async function assertMapsAreEquivalent(a: ISharedMap, b: ISharedMap): Promise<void> {
+	assert.equal(a.size, b.size, `${a.id} and ${b.id} have different number of keys.`);
+	for (const key of a.keys()) {
+		const aVal: unknown = a.get(key);
+		const bVal: unknown = b.get(key);
+		if (isObject(aVal) === true) {
+			assert(
+				isObject(bVal),
+				`${a.id} and ${b.id} differ at ${key}: a is an object, b is not}`,
+			);
+			const aHandle = isFluidHandle(aVal) ? await aVal.get() : aVal;
+			const bHandle = isFluidHandle(bVal) ? await bVal.get() : bVal;
+			assert.equal(
+				aHandle,
+				bHandle,
+				`${a.id} and ${b.id} differ at ${key}: ${JSON.stringify(aHandle)} vs ${JSON.stringify(
+					bHandle,
+				)}`,
+			);
+		} else {
+			assert.equal(aVal, bVal, `${a.id} and ${b.id} differ at ${key}: ${aVal} vs ${bVal}`);
+		}
+	}
+}
+
+const mapReducer = combineReducers<MapOperation, MapState>({
+	clear: ({ client }) => client.channel.clear(),
+	setKey: ({ client }, { key, value }) => {
+		client.channel.set(key, value);
+	},
+	deleteKey: ({ client }, { key }) => {
+		client.channel.delete(key);
+	},
+});
+
+/**
+ * Represents the options for the map generator.
+ */
+interface MapGeneratorOptions {
+	setWeight: number;
+	deleteWeight: number;
+	clearWeight: number;
+	keyPoolSize: number;
+}
+
+const mapDefaultOptions: MapGeneratorOptions = {
+	setWeight: 20,
+	deleteWeight: 20,
+	clearWeight: 1,
+	keyPoolSize: 20,
+};
+
+function mapMakeGenerator(
+	optionsParam?: Partial<MapGeneratorOptions>,
+): AsyncGenerator<MapOperation, MapState> {
+	const { setWeight, deleteWeight, clearWeight, keyPoolSize } = {
+		...mapDefaultOptions,
+		...optionsParam,
+	};
+	// Use numbers as the key names.
+	const keyNames = Array.from({ length: keyPoolSize }, (_, i) => `${i}`);
+
+	const setKey: Generator<MapSetKey, MapState> = ({ random }) => ({
+		type: "setKey",
+		key: random.pick(keyNames),
+		value: random.pick([
+			(): number => random.integer(1, 50),
+			(): string => random.string(random.integer(3, 7)),
+			(): IFluidHandle => random.handle(),
+		])(),
+	});
+	const deleteKey: Generator<MapDeleteKey, MapState> = ({ random }) => ({
+		type: "deleteKey",
+		key: random.pick(keyNames),
+	});
+
+	const syncGenerator = createWeightedGenerator<MapOperation, MapState>([
+		[setKey, setWeight],
+		[deleteKey, deleteWeight],
+		[{ type: "clear" }, clearWeight],
+	]);
+
+	return async (state) => syncGenerator(state);
+}
+
+/**
+ * the maps fuzz model
+ */
+export const baseMapModel: DDSFuzzModel<MapFactory, MapOperation> = {
+	workloadName: "default",
+	factory: new MapFactory(),
+	generatorFactory: () => takeAsync(1000, mapMakeGenerator()),
```

Comment:
```suggestion
	generatorFactory: () => takeAsync(100, mapMakeGenerator()),
```

would be the refactor-only change. I don't feel strongly that we leave it at 100 though.

Reply:
i can revert this for now, as i changes the harness such that it wraps the generator factory such that it can get any number of invocations it needs. we might want to revisit this later